src/general-components/CardComponent/CardComponent.tsx

```diff
@@ -202,6 +257,52 @@ export interface CardComponentProps<D> {
 
 class CardComponent<D = never> extends PureComponent<CardComponentProps<D>, {}> {
 
+    getAllCards = () => {
+        let required = (this.props.required !== undefined) ? this.props.required : true;
+
+        // check and add minimum
+        if (this.props.values.length < this.props.min) {
+            for (let i = 0; i < this.props.min - this.props.values.length; i++) {
```

Comment:
this for loop is useless, multiple calls to addCard don't have any affect, because the internal array of cards is a prop and immutable. We need to change the addCard function so it can add muliple cards at once (for example with a function argument).